homeassistant/components/formula_one/sensor.py

```diff
@@ -0,0 +1,492 @@
+                standing.constructor.name,
+            )
+            if new_cn_sensor.unique_id not in coordinator.f1_data.entity_ids:
+                entities.append(new_cn_sensor)
+
+        for standing in coordinator.f1_data.driver_standings:
+            new_dp_sensor = F1DriverPositionSensor(coordinator, standing.position)
+            if new_dp_sensor.unique_id not in coordinator.f1_data.entity_ids:
+                entities.append(new_dp_sensor)
+
+            new_dn_sensor = F1DriverNameSensor(
+                coordinator,
+                standing.driver.driver_id,
+            )
+            if new_dn_sensor.unique_id not in coordinator.f1_data.entity_ids:
+                entities.append(new_dn_sensor)
+
+        for race in coordinator.f1_data.races:
+            new_race_sensor = F1RaceSensor(coordinator, race.round_no)
+            if new_race_sensor.unique_id not in coordinator.f1_data.entity_ids:
+                entities.append(new_race_sensor)
+
+        next_race_sensor = F1NextRaceSensor(coordinator)
+        if next_race_sensor.unique_id not in coordinator.f1_data.entity_ids:
+            entities.append(next_race_sensor)
+
+        async_add_entities(entities)
+
+    async_setup_sensors()
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, F1_DISCOVERY_NEW, async_setup_sensors)
+    )
+
+
+class F1Sensor(CoordinatorEntity, RestoreSensor):
+    """Base class for a F1 sensor."""
+
+    def _update_value(self) -> None:
+        raise NotImplementedError
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_value()
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Handle event when sensor is added."""
+        await super().async_added_to_hass()
+        f1_data: F1Data = self.coordinator.f1_data
+        if self.unique_id is not None:
+            unique_id: str = self.unique_id
+            f1_data.entity_ids.append(unique_id)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Handle event when entity is removed."""
+        await super().async_will_remove_from_hass()
+        f1_data: F1Data = self.coordinator.f1_data
+        if self.unique_id is not None:
+            unique_id: str = self.unique_id
+            f1_data.entity_ids.remove(unique_id)
+
+
+class F1ConstructorPositionSensor(F1Sensor):
+    """Representation of a F1 sensor."""
+
+    def __init__(
+        self,
+        coordinator: F1UpdateCoordinator,
+        position: int,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+
+        self.position = position
+        self._update_value()
+
+    def _update_value(self) -> None:
+        standing_info = self.coordinator.f1_data.get_constructor_standing_by_position(
+            self.position
+        )
+
+        if standing_info is None:
```

Comment:
What is enabling the sensor again once is becomes available later on?

Reply:
This was a bug.